src/core/manager.c

```diff
@@ -1663,7 +1663,11 @@ static void manager_invoke_notify_message(Manager *m, Unit *u, pid_t pid, const
         assert(m);
         assert(u);
         assert(buf);
-        assert(n > 0);
+
+        if(n <= 0) {
+            log_unit_debug(u, "Got a zero-length notification message. Ignoring");
+            return;
+        }
```

Comment:
This function is only called from manager_dispatch_notify_fd(), which you fix below. So an assert is appropriate here, we don't want to check the same condition twice.